### PR TITLE
feat: setup typescript, prettier, eslint and basic structure

### LINKED ISSUE #2: Create the static data source & the main class in repository

### PR DESCRIPTION
**Pull request by aabishkaryal**

### CODE REVIEW

File: src/data/days-in-month-mapping.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
+export default {
```
Comment (ashiishme) [inline src/data/days-in-month-mapping.ts]: Can you rather make the named export?

File: src/data/nepali-days.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+export default [
```
Comment (ashiishme) [inline src/data/nepali-days.ts]: Same here and applies to all other default exports of data.

File: src/data/nepali-months.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+export default [
```
Comment (ashiishme) [inline src/data/nepali-months.ts]: 👉

File: src/data/new-year-mapping.ts [hunk 4]
```diff
@@ -0,0 +1,133 @@
+export default {
```
Comment (ashiishme) [inline src/data/new-year-mapping.ts]: 👉

File: src/bikramSambat.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import NepaliDays from "data/nepali-days";
+import NepaliMonths from "data/nepali-months";
+import NewYearMapping from "data/new-year-mapping";
+import DaysInMonthMapping from "data/days-in-month-mapping";
+
+export default class BikramSambat {
```
Comment (ashiishme) [inline src/bikramSambat.ts]: The file name should be `BikramSambat.ts`

**Review by ashiishme — CHANGES_REQUESTED** [review]
Also, we can put the `data` directory outside of `src`.

**Review by ashiishme — APPROVED** [review]

File: src/BikramSambat.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import { NepaliDaysData } from "data/nepali-days";
+import { NepaliMonthsData } from "data/nepali-months";
+import { NewYearMappingData } from "data/new-year-mapping";
+import { DaysInMonthsMappingData } from "data/days-in-month-mapping";
```
Comment (blackkspydo) [inline src/BikramSambat.ts]: suggestion: what do you say if we export all data objects from single `index.ts` file in data folder, that way we will have single point of data imports and no worry of messing up of import paths

**Review by blackkspydo — APPROVED** [review]

**Pull request merged**